pkg/services/dashboards/folder_service.go

```diff
@@ -61,7 +61,7 @@ func (dr *dashboardServiceImpl) GetFolders(limit int64) ([]*models.Folder, error
 }
 
 func (dr *dashboardServiceImpl) GetFolderByID(id int64) (*models.Folder, error) {
-	query := models.GetDashboardQuery{OrgId: dr.orgId, Id: id}
+	query := models.GetDashboardQuery{OrgId: dr.orgId, Id: id, IsFolder: true}
```

Comment:
Wonder what is expected a 404 folder not found or a 200 response with only the id and the title populated in the response? The general folder is not stored in the database so doesn't make sense in this context to even query the database if the id is less than or equal zero. A potential alternative would be to handle `models.ErrDashboardIdentifierNotSet` in `toFolderError`.

The `GetDashboardQuery` is already quite easy to get wrong why adding one additional field might not be a good idea.

Reply:
> Wonder what is expected a 404 folder not found or a 200 response with only the id and the title populated in the response?

From a user experience perspective, the General folder *does* exist, but is read-only. So the API giving a 404 is confusing. That it doesn't exist in the database is an implementation detail. Hence, my suggestion here is to return a simple 200 response to a GET request, an to a PUT or a POST give a "not allowed, read only" type response.

Reply:
@malcolmholmes , made some changes, so now after calling `api/folders/id/0`, we are getting:

```
{"id":0,"uid":"","title":"General","url":"","hasAcl":false,"canSave":true,"canEdit":true,"canAdmin":true,"createdBy":"Anonymous","created":"0001-01-01T00:00:00Z","updatedBy":"Anonymous","updated":"0001-01-01T00:00:00Z","version":0}
```

I think that we cannot have PUT or POST requests with an id - only Uid AFAICT.

@marefr WDYT?